src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -325,6 +327,30 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         });
     }
 
+    private async updateProjectTagsFromAsset(asset: IAssetMetadata) {
+        const assetTags = new Set();
+        asset.regions.forEach((region) => region.tags.forEach((tag) => assetTags.add(tag.name)));
+
+        const newTags: ITag[] = this.props.project.tags ? this.props.project.tags.slice() : [];
+        let updateTags = false;
+
+        assetTags.forEach((tag) => {
+            if (!this.props.project.tags || this.props.project.tags.length === 0 ||
+                this.props.project.tags.filter((projectTag) => tag === projectTag.name).length === 0 ) {
+                newTags.push({
+                    name: tag,
+                    color: "",
```

Comment:
We should reuse the `tagsInput` method to generate a color for this newly created tag.

Reply:
I just wonder why we have color in project.tags (ITag).  It seams a duplication of info as we don't really have random color or any way to let the user specify a color.  It should just be the correspondence position in the TagColors collection.

Reply:
The user can change the tag color by `Ctrl+Click` on the tag in either the editor or properties page.